### PR TITLE
Fix SpanWrapper toString to avoid duplicates

### PR DESCRIPTION
**Pull request by bogdandrutu**
Fixes https://github.com/open-telemetry/opentelemetry-java/issues/2613

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: sdk/trace/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java [hunk 1]
```diff
@@ -497,45 +497,4 @@ private Attributes getImmutableAttributes() {
     // otherwise, make a copy of the data into an immutable container.
     return attributes.immutableCopy();
   }
-
-  @Override
-  public String toString() {
```
Comment (jkwatson) [inline sdk/trace/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java:502]: do we really want to get rid of this toString()?  We can fix the other one without getting rid of this one.

File: sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SpanWrapper.java [hunk 2]
```diff
@@ -158,4 +158,57 @@ public int getTotalRecordedLinks() {
   public int getTotalAttributeCount() {
     return totalAttributeCount();
   }
+
+  @Override
+  public final String toString() {
```
Comment (jkwatson) [inline sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SpanWrapper.java:163]: final here? Seems unnecessarily restrictive.

Comment (jkwatson) [reply sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SpanWrapper.java:163]: oh, does this make autoValue not try to implement it? I forgot about that. nm

**Review by kenfinnigan — APPROVED** [review]

File: sdk/trace/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java (hunk 1 above)
Comment (bogdandrutu) [reply sdk/trace/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java:502]: I don't want users to call `toString` on the `Span` and parse the String to extract fields, they should do that only in the processor via the `ReadableSpan`.

Comment (anuraaga) [reply sdk/trace/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java:502]: `toString` is important in debugging, e.g., it's presented well by the IDE. We should definitely not be removing `toString`'s we already have. While it'd be unfortunate if anyone parses the tostring, that's a bit too explicit foot-shooting to lose the debugging experience for normal cases.

Comment (jkwatson) [reply sdk/trace/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java:502]: Agree with @anuraaga . Let's not get rid of valuable debugging information, even if a user might do something unsupported. They can also use reflection if they want to do bad things to themselves.

Comment (bogdandrutu) [reply sdk/trace/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java:502]: I will move this to a separate PR and still fight it. I am not sure what debugging scenario you have in mind

Comment (bogdandrutu) [reply sdk/trace/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java:502]: Reverted.

**Comment by bogdandrutu** [comment]
For some reasons this PR does not get updated, if you look at my branch https://github.com/open-telemetry/opentelemetry-java/compare/main...bogdandrutu:fixsdtostring the removing of the toString was reverted, I will give few more minutes to see if github picks this up

**Review by jkwatson — APPROVED** [review]

**Review by carlosalberto — APPROVED** [review]

**Pull request merged**